dbms/src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -89,10 +89,46 @@ std::optional<String> findFirstNonDeterministicFuncName(const MutationCommand &
 
     return {};
 }
-};
 
+ASTPtr prepareQueryAffectedAST(const std::vector<MutationCommand> & commands)
+{
+    /// Execute `SELECT count() FROM storage WHERE predicate1 OR predicate2 OR ...` query.
+    /// The result can differ from tne number of affected rows (e.g. if there is an UPDATE command that
+    /// changes how many rows satisfy the predicates of the subsequent commands).
```

Comment:
Is difficult to understand. Maybe add an example?
If I am not mistaken, the idea is simple: not all the rows we read are satisfy the predicates.